server_spec.rb

```diff
@@ -0,0 +1,234 @@
+      subject(:id) { JSON.parse(response.body)['message']['id'] }
+
+      context 'valid params' do
+        specify { expect(response.status).to eq(200) }
+        specify 'creates a message db record' do
+          expect(message_record_with?(db: @db, id: id, attribute: :to_user, value: attributes[:to])).to be true
+          expect(message_record_with?(db: @db, id: id, attribute: :from_user, value: attributes[:from])).to be true
+          expect(message_record_with?(db: @db, id: id, attribute: :message, value: attributes[:message])).to be true
+        end
+        specify 'enqueues the message to be sent' do
+          response
+          expect(described_class::ExternalApiClient).to have_received(:send_message).with(attributes)
+        end
+      end
+
+      context 'invalid params' do
+        let(:attributes) do
+          {
+            to: to_id,
+            from: 'non-existant-id',
+            message: 'this is a test'
+          }
+        end
+        specify "doesn't create a message db record" do
+          expect(no_message_record?(@db)).to be true
+        end
+        specify { expect(response.status).to eq(400) }
+      end
+    end
+
+    context 'xml' do
+      let(:format) { 'xml' }
+      let(:request_body) do
+        builder = Nokogiri::XML::Builder.new do |xml|
+          xml.root do
+            xml.to attributes[:to]
+            xml.from attributes[:from]
+            xml.message attributes[:message]
+          end
+        end
+        builder.to_xml
+      end
+
+      subject(:id) { Nokogiri.XML(response.body).xpath("//message/id").text }
+
+      context 'valid params' do
+        specify { expect(response.status).to eq(200) }
+        specify 'creates a message db record' do
+          expect(message_record_with?(db: @db, id: id, attribute: :to_user, value: attributes[:to])).to be true
+          expect(message_record_with?(db: @db, id: id, attribute: :from_user, value: attributes[:from])).to be true
+          expect(message_record_with?(db: @db, id: id, attribute: :message, value: attributes[:message])).to be true
+        end
+        specify 'enqueues the message to be sent' do
+          response
+          expect(described_class::ExternalApiClient).to have_received(:send_message).with(attributes)
+        end
+      end
+
+      context 'invalid params' do
+        let(:attributes) do
+          {
+            to: to_id,
+            from: 'non-existant-id',
+            message: 'this is a test'
+          }
+        end
+        specify "doesn't create a message db record" do
+          expect(no_message_record?(@db)).to be true
+        end
+        specify { expect(response.status).to eq(400) }
+      end
+    end
+
+    context 'other' do
+      let(:format) { 'csv' }
+      let(:request_body) { '' }
+      specify { expect(response.status).to eq(400) }
+    end
+  end
+
+  describe 'GET /messages/:id/status' do
+    let(:id) { SecureRandom.uuid }
+    let(:status) { 'sent' }
+    before do
```

Comment:
Given the duplication of the database setup, I think we should group all the test contexts that interact with the DB under one context and move the setup logic there.